src/_P103_Atlas_EZO_pH_ORP_EC.ino

```diff
@@ -0,0 +1,641 @@
+            } else {
+                addHtml(F("&nbsp;<span style='color:red;'>Not yet calibrated</span>"));
+            }
+            addFormCheckBox(F("Enable"), F("Plugin_103_enable_cal_M"), false);
+            addHtml(F(
+                        "\n<script type='text/javascript'>document.getElementById(\"Plugin_103_enable_cal_M\").onclick = function(){document.getElementById(\"Plugin_103_enable_cal_L\").checked = false;document.getElementById(\"Plugin_103_enable_cal_H\").checked = false;};</script>\n"));
+
+            addRowLabel(F("<strong>Low</strong>"));
+            addFormFloatNumberBox(F("Ref Ph"),
+                                    F("Plugin_103_ref_cal_L"),
+                                    PCONFIG_FLOAT(2),
+                                    0,
+                                    14,
+                                    2,
+                                    0.01);
+
+            if (nb_calibration_points > 1) {
+                addHtml(F("&nbsp;<span style='color:green;'>OK</span>"));
+            } else {
+                addHtml(F("&nbsp;<span style='color:red;'>Not yet calibrated</span>"));
+            }
+            addFormCheckBox(F("Enable"), F("Plugin_103_enable_cal_L"), false);
+            addHtml(F(
+                        "\n<script type='text/javascript'>document.getElementById(\"Plugin_103_enable_cal_L\").onclick = function(){document.getElementById(\"Plugin_103_enable_cal_M\").checked = false;document.getElementById(\"Plugin_103_enable_cal_H\").checked = false;};</script>\n"));
+
+            addHtml(F("<TR><TD><strong>High</strong></TD>"));
+            addFormFloatNumberBox(F("Ref Ph"),
+                                    F("Plugin_103_ref_cal_H"),
+                                    PCONFIG_FLOAT(3),
+                                    0,
+                                    14,
+                                    2,
+                                    0.01);
+
+            if (nb_calibration_points > 2) {
+                addHtml(F("&nbsp;<span style='color:green;'>OK</span>"));
+            } else {
+                addHtml(F("&nbsp;<span style='color:orange;'>Not yet calibrated</span>"));
+            }
+            addFormCheckBox(F("Enable"), F("Plugin_103_enable_cal_H"), false);
+            addHtml(F(
+                        "\n<script type='text/javascript'>document.getElementById(\"Plugin_103_enable_cal_H\").onclick = function(){document.getElementById(\"Plugin_103_enable_cal_L\").checked = false;document.getElementById(\"Plugin_103_enable_cal_M\").checked = false;};</script>\n"));
+
+            if (nb_calibration_points > 1) {
+                char sensordata[32];
+                bool status;
+                status = _P103_send_I2C_command(I2Cchoice, F("Slope,?"), sensordata);
+
+                if (status) {
+                String slopeAnswer = F("Answer to 'Slope' command : ");
+                slopeAnswer += sensordata;
```

Comment:
Is `sensordata` zero-terminated? (guaranteed!!!)

Reply:
I don't know. I will check this.

Reply:
now sensordata always will be zero terminated due to several thing: initalization with {0} and zero termination and check in i2c function